makefiles/test.mk

```diff
@@ -16,29 +16,73 @@ test_unit: ## Run all unit tests
+	@if [ "$(filter-out $@,$(MAKECMDGOALS))" = "" ]; then \
+		echo "❌ Error: Service IDs are required (comma-separated list)"; \
+		echo "  👀 Example: make test_e2e_evm_morse F00C,F021"; \
+		echo "  💡 To run with default service IDs, use: make test_e2e_evm_morse_defaults"; \
+		exit 1; \
+	fi
+	(cd e2e && TEST_MODE=e2e TEST_PROTOCOL=morse TEST_SERVICE_IDS=$(filter-out $@,$(MAKECMDGOALS)) go test -v -tags=e2e -count=1 -run Test_PATH_E2E_EVM)
+
+.PHONY: test_e2e_evm_shannon_defaults
+test_e2e_evm_shannon_defaults: shannon_e2e_config_warning ## Run an E2E Shannon relay test with default service IDs
 	(cd e2e && TEST_MODE=e2e TEST_PROTOCOL=shannon go test -v -tags=e2e -count=1 -run Test_PATH_E2E_EVM)
 
+.PHONY: test_e2e_evm_shannon
+test_e2e_evm_shannon: shannon_e2e_config_warning ## Run an E2E Shannon relay test with specified service IDs (e.g. make test_e2e_evm_shannon eth,anvil)
+	@if [ "$(filter-out $@,$(MAKECMDGOALS))" = "" ]; then \
+		echo "❌ Error: Service IDs are required (comma-separated list)"; \
+		echo "  👀 Example: make test_e2e_evm_shannon eth,anvil"; \
+		echo "  💡 To run with default service IDs, use: make test_e2e_evm_shannon_defaults"; \
+		exit 1; \
+	fi
+	(cd e2e && TEST_MODE=e2e TEST_PROTOCOL=shannon TEST_SERVICE_IDS=$(filter-out $@,$(MAKECMDGOALS)) go test -v -tags=e2e -count=1 -run Test_PATH_E2E_EVM)
+
 ##################
 ### Load Tests ###
 ##################
 
-.PHONY: test_load_evm_morse
-test_load_evm_morse: morse_e2e_config_warning ## Run a Morse load test
+.PHONY: test_load_evm_morse_defaults
+test_load_evm_morse_defaults: morse_e2e_config_warning ## Run a Morse load test with default service IDs
 	(cd e2e && TEST_MODE=load TEST_PROTOCOL=morse go test -v -tags=e2e -count=1 -run Test_PATH_E2E_EVM)
 
-.PHONY: test_load_evm_shannon
-test_load_evm_shannon: shannon_e2e_config_warning ## Run a Shannon load test
+.PHONY: test_load_evm_morse
+test_load_evm_morse: morse_e2e_config_warning ## Run a Morse load test with specified service IDs (e.g. make test_load_evm_morse F00C,F021)
+	@if [ "$(filter-out $@,$(MAKECMDGOALS))" = "" ]; then \
+		echo "❌ Error: Service IDs are required (comma-separated list)"; \
+		echo "  👀 Example: make test_load_evm_morse F00C,F021"; \
+		echo "  💡 To run with default service IDs, use: make test_load_evm_morse_defaults"; \
+		exit 1; \
+	fi
+	(cd e2e && TEST_MODE=load TEST_PROTOCOL=morse TEST_SERVICE_IDS=$(filter-out $@,$(MAKECMDGOALS)) go test -v -tags=e2e -count=1 -run Test_PATH_E2E_EVM)
+
+.PHONY: test_load_evm_shannon_defaults
+test_load_evm_shannon_defaults: shannon_e2e_config_warning ## Run a Shannon load test with default service IDs
 	(cd e2e && TEST_MODE=load TEST_PROTOCOL=shannon go test -v -tags=e2e -count=1 -run Test_PATH_E2E_EVM)
 
+.PHONY: test_load_evm_shannon
+test_load_evm_shannon: shannon_e2e_config_warning ## Run a Shannon load test with specified service IDs (e.g. make test_load_evm_shannon eth,anvil)
+	@if [ "$(filter-out $@,$(MAKECMDGOALS))" = "" ]; then \
```

Comment:
If we require an env var (no defaults), we can simplify this.

Reply:
It's been updated to require the service ID.